Makefile.am

```diff
@@ -57,6 +57,18 @@ spellcheck spellcheck-interactive:
 doc spellcheck-sortdict:
 	cd $(srcdir)/docs && $(MAKE) $@
 
+# This target adds syntax-checking for committed shell script files,
+# to avoid surprises and delays in finding fatal typos after packaging
+check-scripts-syntax shellcheck:
+	@echo 'NOTE: modern bash complains about scripts using backticks (which we ignore for portability reasons): `...` obsolete, use $(...)'
+	for F in `git ls-files || find . -type f` ; do \
+	    case "`file "$$F"`" in \
+	        *"Bourne-Again shell script"*) ( set -x ; /bin/bash -n "$$F" ; ) ;; \
+	        *"POSIX shell script"*|*"shell script"*) ( set -x ; /bin/sh -n "$$F" ; ) ;; \
```

Comment:
I don't mean to nit-pick, but it seems as though "shell script" is a superset of "POSIX shell script". Did you intend to have another option here?